realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy

```diff
@@ -0,0 +1,181 @@
+                it.getAnnotations()
+                it.getAnnotation(Ignore.class) == null && !Modifier.isStatic(it.getModifiers())
+            })
+        }
+        logger.info "Managed Fields: ${managedFields*.name}"
+
+        // Add accessors to the model classes
+        modelClasses.each {
+            BytecodeModifier.addRealmAccessors(it)
+            def proxyInterface = classPool.get("io.realm.${it.getSimpleName()}RealmProxyInterface")
+            it.addInterface(proxyInterface)
+        }
+
+        // Use accessors instead of direct field access
+        classNames.each {
+            logger.info "  Modifying class ${it}"
+            def ctClass = classPool.getCtClass(it)
+            BytecodeModifier.useRealmAccessors(ctClass, managedFields, modelClasses)
+            ctClass.writeFile(outputProvider.getContentLocation(
+                    'realm', getInputTypes(), getScopes(), Format.DIRECTORY).canonicalPath)
+        }
+
+        def toc = System.currentTimeMillis()
+        logger.info "Realm Transform time: ${toc-tic} milliseconds"
+    }
+
+    /**
+     * Create and populate the Javassist class pool
+     * @param inputs The inputs provided by the Transform API
+     * @param referencedInputs the referencedInputs provided by the Transform API
+     * @return the populated ClassPool instance
+     */
+    private ClassPool createClassPool(Collection<TransformInput> inputs, Collection<TransformInput> referencedInputs) {
+        // Don't use ClassPool.getDefault(). Doing consecutive builds in the same run (e.g. debug+release)
+        // will use a cached object and all the classes will result frozen.
+        ClassPool classPool = new ClassPool(null)
+        classPool.appendSystemPath()
+        classPool.appendClassPath(new LoaderClassPath(getClass().getClassLoader()))
+
+        inputs.each {
+            it.directoryInputs.each {
+                classPool.appendClassPath(it.file.absolutePath)
+            }
+
+            it.jarInputs.stream().each {
+                classPool.appendClassPath(it.file.absolutePath)
+            }
+        }
+
+        referencedInputs.each {
+            it.directoryInputs.each {
+                classPool.appendClassPath(it.file.absolutePath)
+            }
+
+            it.jarInputs.stream().each {
+                classPool.appendClassPath(it.file.absolutePath)
+            }
+        }
+
+        return classPool
+    }
+
+    private Set<String> getClassNames(Collection<TransformInput> inputs) {
+        Set<String> classNames = new HashSet<String>()
+
+        inputs.each {
+            it.directoryInputs.each {
+                def dirPath = it.file.absolutePath
+                it.file.eachFileRecurse(FileType.FILES) {
+                    if (it.absolutePath.endsWith(SdkConstants.DOT_CLASS)) {
+                        def className = it.absolutePath.substring(dirPath.length() + 1, it.absolutePath.length() - SdkConstants.DOT_CLASS.length()).replaceAll('/', '.')
```

Comment:
Right. I was thinking of having to deal with stuff like `C:\` but that should have been stripped before already